helix-view/src/document.rs

```diff
@@ -1210,6 +1218,12 @@ impl Document {
                     .ensure_invariants(self.text.slice(..));
             }
 
+            for view_data in self.view_data.values_mut() {
+                view_data.view_position.anchor = transaction
+                    .changes()
+                    .map_pos(view_data.view_position.anchor, Assoc::Before);
```

Comment:
I do wonder if we should be using `Assoc::After` here instead. I don't have any opinion here. I will try with before and see if it annoys me. It sprobably doesn't make too much difference due to `ensure_cursor_in_view` keeping the view position somewhat fixed in extreme cases